reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/tests/FeatureScopeTest.kt

```diff
@@ -156,6 +156,49 @@ class FeatureScopeTest : MockServerTest() {
         }.doWait(LONG_WAIT_MS)
     }
 
+    @Test
+    fun mustReceiveTheEvents_whenFeatureWrite_trackingConsentPendingToGranted_switched_asynchronously() {
```

Comment:
By curiosity I remove `@Volatile` that you've added and run the test on my end, the test still passes, so I am wondering what's the case are you exactly testing against? are you testing the thread-safety when `delegateOrchestrator` is assigned in `ConsentAwareFileOrchestrator`? is there any false positive in this unit test?